content/en/docs/KRE/user/60_kais_runner_sdk/_index.md

```diff
@@ -0,0 +1,266 @@
+---
+title: "KAI Server Runner SDK"
+linkTitle: "KAI Server Runner SDK"
+weight: 60
+description: >
+  Code your nodes on top of the KAI Server Runner SDK
+---
+
+
+## The Runner SDK
+
+KAI Server supports Python and Golang providing an SDK that allows users to receive,
+process and send messages between nodes.
+
+Thanks to the SDK, users can implement `handler` and `init` functions.
+
+## Init function
+
+This function receives a context that provides methods for logging and storing data that users
+can later use on the handler function. It doesn't return any value.
+
+`init` function is mainly used to initialize data, for example, loading models.
+
+### Python init example
+
+```python
+import joblib
+
+async def init(ctx):
+  ctx.logger.info("[worker init]")
+  ctx.set('model', joblib.load(ctx.path('models/model.joblib')))
+```
+
+### Golang init example
+
+```golang
+package main
+
+import (
+  "github.com/konstellation-io/kre/runners/kre-go"
+)
+
+func handlerInit(ctx *kre.HandlerContext) {
+  ctx.Logger.Info("init handler")
+    
+  // Saves a value in the context internal registry
+  ctx.Set("greeting", "Hello")   
+}
+```
+
+## Handler function
+
+This function receives a context, the same as the `init` function does, and the data, which is the incoming payload to process sent from this node's subscription. The context object is shared between different executions.
+
+`Handler` functions are used to process data, users can implement their service logic here.
+
+### Python handler example for runner compatible with KRT V1
+
+```python
+import pandas as pd
+
+async def handler(ctx, data):
+    ctx.logger.info("message received")
+
+    # Create a dataframe from input dict
+    df = pd.DataFrame.from_dict(data)
+
+    # Access preloaded data from context loaded on init() function 
+    model = ctx.get('model')
+
+    # Run a prediction
+    ctx.logger.info('running prediction')
+    prediction = model.predict(df)
+
+    # Return prediction value as a JSON serializable object
+    return {'price_category': prediction.item()}
+```
+
+### Golang handler example for runner compatible with KRT V1
+
+```golang
+func handler(ctx *kre.HandlerContext, data *any.Any) (proto.Message, error) {
+ ctx.Logger.Info("[handler invoked]")
+
+ req := &Request{}
+ res := &EtlOutput{}
+
+ err := anypb.UnmarshalTo(data, req, proto2.UnmarshalOptions{})
+ if err != nil {
+  return res, fmt.Errorf("invalid request: %s", err)
+ }
+
+ res.Component = req.Component
+
+ saveEtlMetrics(ctx, req.Component)
+
+ return res, nil
+}
+```
+
+## How to provide node functions to kre
+
+Once the `init` and `handler` functions have been declared and implemented, they have to be served in a certain way, so runners recognize them.
+
+### For runners compatible with KRT V1
+
+#### Python (images v1 and v2)
```

Comment:
Imo, I wouldn't indicate here these image tags. They are not relevant in this part of the documentation, and it should exist a compatibility table instead.

Reply:
Good idea about the compatibility table. I think as well it could be clearer if we were just to group them in krt version compatibility.